pkg/operators/extractors/compile.go

```diff
@@ -4,12 +4,23 @@ import (
 	"fmt"
 	"regexp"
 	"strings"
+	"sync"
+	"sync/atomic"
 
 	"github.com/Knetic/govaluate"
 	"github.com/itchyny/gojq"
 	"github.com/projectdiscovery/nuclei/v3/pkg/operators/common/dsl"
 )
 
+var (
+	extractorRegexCache        sync.Map // map[string]*regexp.Regexp
+	extractorDslCache          sync.Map // map[string]*govaluate.EvaluableExpression
+	extractorMaxRegexCacheSize = 4096
```

Comment:
why are we setting max cache size? we should also look into how much the default nuclei templates occupy the cache so we can choose a better number for max performance rather than arbitrary